File: src/types/label.ts
export interface Label {
  id: number;
  name: string;
  color?: string;
  sortOrder: number;
}

export interface CreateLabel {
  name: string;
  color?: string;
}

export interface UpdateLabel {
  id: number;
  name?: string;
  color?: string;
  sortOrder?: number;
}

File: src-tauri/src/models/label.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Label {
    pub id: i64,
    pub name: String,
    pub color: Option<String>,
    #[serde(rename = "sortOrder")]
    pub sort_order: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateLabel {
    pub name: String,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateLabel {
    pub id: i64,
    pub name: Option<String>,
    pub color: Option<String>,
    pub sort_order: Option<i64>,
}
